src/sentry/models/organizationoption.py

```diff
@@ -1,30 +1,14 @@
 from __future__ import absolute_import, print_function
 
-from celery.signals import task_postrun
-from django.core.signals import request_finished
 from django.db import models
 
 from sentry.db.models import Model, FlexibleForeignKey, sane_repr
 from sentry.db.models.fields import EncryptedPickledObjectField
-from sentry.db.models.manager import BaseManager
+from sentry.db.models.manager import OptionManager
 from sentry.utils.cache import cache
 
 
-class OrganizationOptionManager(BaseManager):
-    def __init__(self, *args, **kwargs):
-        super(OrganizationOptionManager, self).__init__(*args, **kwargs)
-        self.__cache = {}
-
-    def __getstate__(self):
-        d = self.__dict__.copy()
-        # we cant serialize weakrefs
-        d.pop("_OrganizationOptionManager__cache", None)
-        return d
-
-    def __setstate__(self, state):
```

Comment:
Do we need these in `OptionManager` now? Probably need to be careful not to shove this state into a pickle

Reply:
Modified things a bit by sharing the cache between the options models. The newer implementation should mean that we now won't need these.

Reply:
How come though? Isn't the problem that we don't want to store this cache when we pickle objects?

Actually, could you just pickling/unpickling a model that has this cache on it? And confirm that the cache isn't persisted through the pickle?

Reply:
Ah I moved the `cache object` off the `Manager object` entirely which is why this is no longer needed. (Tested, and the [pickle](https://gist.github.com/wmak/81eba331dd8adfebeabdaf944a7c2120) doesn't have the cache)
But you're right in that if the cache was still a part of the manager we'd have to unset them so we don't pickle our cache.

see: https://github.com/getsentry/sentry/pull/15990/files#diff-2489d6a0ce491bf3d0a62314f1715ecfR459-R463